src/main/java/net/testudobank/MvcController.java

```diff
@@ -381,64 +381,28 @@ public String submitDispute(@ModelAttribute("user") User user) {
     int userOverdraftBalanceInPennies = TestudoBankRepository.getCustomerOverdraftBalanceInPennies(jdbcTemplate, userID);
 
     int reversalAmountInPennies = (int) logToReverse.get("Amount");
+    int reversalAmount = reversalAmountInPennies / 100;
 
     // If transaction to reverse is a deposit, then withdraw the money out
     if (((String) logToReverse.get("Action")).toLowerCase().equals("deposit")) {
       // if withdraw would exceed max overdraft possible, return welcome
       if (userOverdraftBalanceInPennies + (reversalAmountInPennies - userBalanceInPennies) > MAX_OVERDRAFT_IN_PENNIES) {
         return "welcome";
       }
+      user.setAmountToWithdraw(reversalAmount);
+      submitWithdraw(user);
 
-      // if balance is large enough to have reversalAmount taken from it, subtract reversalAmount from balance
-      if (userBalanceInPennies - reversalAmountInPennies > 0){
-        TestudoBankRepository.decreaseCustomerBalance(jdbcTemplate, userID, reversalAmountInPennies);
-      } else { // Case when reversing deposit causes overdraft or go deeper into overdraft
-        // Set main balance to 0 since we are either going into overdraft or already in overdraft
-        TestudoBankRepository.setCustomerBalance(jdbcTemplate, userID, 0);
-
-        int differenceInPennies = reversalAmountInPennies - userBalanceInPennies;
-
+      if (userBalanceInPennies - reversalAmountInPennies <= 0){
```

Comment:
Re-write this to `if (reversalAmountInPennies > userBalanceInPennies)` so that it is easier to read (your readers don't need to do a subtraction and understand why that difference being <= 0 is significant). Also, it should be `>`, not `>=`. Having the withdraw amount (`reversalAmountInPennies`) be equal to the user's balance will just make the user's new balance $0, and won't put them in overdraft.

Also, add a comment at the top of the if-statement to make it clear that is resolving the case where the reversed deposit is putting the customer back into overdraft.

There are 2 possibilities at this point that are covered in the logic within the body of this if-statement: the deposit that was reversed contributed to an overdraft repayment that must have the repayment log erased, OR the deposit was just a normal deposit when the customer was not in overdraft (no repayment log erasing needed).

Reply:
This makes sense, I have changed the statement to improve readability. I have also added a comment to make it clear which case I am handling.